components/Form/Edit.js

```diff
@@ -0,0 +1,84 @@
+import { useState } from "react";
+import { FormContainer, FormButton, AddButton } from "./styles-form";
+import { Heading } from "./styles-form";
+
+export default function EditRecipe({ initialRecipe, onSubmit }) {
+  const [recipe, setRecipe] = useState(initialRecipe);
+
+  const handleInputChange = (event) => {
+    const { name, value } = event.target;
+    setRecipe((prevRecipe) => ({
+      ...prevRecipe,
+      [name]: value,
+    }));
+  };
+
+  const handleIngredientChange = (index, field, event) => {
+    const updatedIngredients = [...recipe.ingredients];
+    updatedIngredients[index][field] = event.target.value;
+
+    setRecipe((prevRecipe) => ({
+      ...prevRecipe,
+      ingredients: updatedIngredients,
+    }));
+  };
+
+  const addIngredient = () => {
+    setRecipe((prevRecipe) => ({
+      ...prevRecipe,
+      ingredients: [...prevRecipe.ingredients, { name: "", quantity: "" }],
+    }));
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    onSubmit(recipe);
+  };
+
+  return (
+    <>
+      <Heading>Edit Recipe</Heading>
+      <FormContainer onSubmit={handleSubmit}>
+        <input
+          type="text"
+          name="name"
+          placeholder="Recipe Name"
+          value={recipe.name}
+          onChange={handleInputChange}
+          required
+        />
+
+        {recipe.ingredients.map((ingredient, index) => (
+          <div key={index}>
```

Comment:
`index` is not enough for the `key` prop. You need a stable id or something else that is unique to your data.